File: backend/src/main/java/com/wizardom/backend/domain/university/model/University.java
package com.wizardom.backend.domain.university.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.wizardom.backend.domain.university.department.model.Department;
import com.wizardom.backend.domain.students.model.Student;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import lombok.experimental.Accessors;

import javax.persistence.*;
import java.util.List;

@Accessors(chain = true)
@Entity(name = "universities")
@Getter
@Setter
@ToString
public class University {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    private String shortName;
    private String fullName;

    @OneToMany
    private List<Department> departments;

    @JsonIgnore
    @OneToMany(mappedBy = "university")
    private List<Student> students;

}

File: app/lib/application/university/university.dart
import 'package:app/application/university/department.dart';

class University {
  final int id;
  final String shortName;
  final String fullName;
  final List<Department> department;

  University({this.id, this.shortName, this.fullName, this.department});

  factory University.fromServer(dynamic data) {
    return University(
      id: data["id"],
      shortName: data["shortName"],
      fullName: data["fullName"],
    );
  }

  @override
  String toString() {
    return 'University{id: $id, shortName: $shortName, fullName: $fullName, department: $department}';
  }
}
